core-processor/src/precharge.rs

```diff
@@ -421,27 +406,22 @@ pub fn precharge_for_memory(
         }
         Err(err) => {
             log::debug!("Failed to charge for module instantiation or memory pages: {err:?}");
-            let reason = match err {
-                PrechargeError::BlockGasExceeded => {
-                    return Err(process_allowance_exceed(
-                        context.data.dispatch,
-                        context.data.destination_id,
-                        context.data.gas_counter.burned(),
-                    ));
-                }
-                PrechargeError::GasExceeded(op) => {
-                    ActorExecutionErrorReplyReason::PreChargeGasLimitExceeded(op)
-                }
+            let PrechargeError::GasExceeded(op) = err else {
+                return Err(process_allowance_exceed(
+                    context.data.dispatch,
+                    context.data.destination_id,
+                    context.data.gas_counter.burned(),
+                ));
             };
 
             let system_reservation_ctx =
                 SystemReservationContext::from_dispatch(&context.data.dispatch);
-            return Err(process_error(
+            return Err(process_execution_error(
                 context.data.dispatch,
                 context.data.destination_id,
                 context.data.gas_counter.burned(),
                 system_reservation_ctx,
-                reason,
+                ActorExecutionErrorReplyReason::PreChargeGasLimitExceeded(op),
                 false,
```

Comment:
why false

Reply:
this is a bug, thank you

Reply:
must be true

Reply:
removed, now we send signal always when we can do it